http/oidc/src/main/java/org/wildfly/security/http/oidc/TokenValidator.java

```diff
@@ -83,13 +83,16 @@ private TokenValidator(Builder builder) {
      * @return the {@code VerifiedTokens} if the ID token was valid
      * @throws OidcException if the ID token is invalid
      */
-    public VerifiedTokens parseAndVerifyToken(final String idToken, final String accessToken, OidcHttpFacade.Cookie cookie) throws OidcException {
+    public VerifiedTokens parseAndVerifyToken(final String idToken, final String accessToken,
```

Comment:
I think we can also have the three argument method, i.e., `parseAndVerifyToken(final String idToken, final String accessToken, OidcHttpFacade.Cookie cookie)` that just calls the 4-arg version with `validateNonce` set to `false`, i.e.,

```
public VerifiedTokens parseAndVerifyToken(final String idToken, final String accessToken, OidcHttpFacade.Cookie cookie) throws OidcException {
    return parseAndVerifyToken(idToken, accessToken, cookie, false);
}
```

Then, we can call the 4-arg version explicitly for the initial authentication request case. WDYT?

Reply:
fixed